modules/Bio/EnsEMBL/DBSQL/SliceAdaptor.pm

```diff
@@ -1113,7 +1145,7 @@ sub fetch_all {
 
   my @out;
   while($sth->fetch()) {
-    if(!defined($bad_vals{$seq_region_id})){
+    if(!defined($bad_vals{$seq_region_id}) && (!$only_protein_coding || $good_vals{$seq_region_id})){
```

Comment:
Should use exists here to prevent pointless hash growth over potentially large numbers of results.

Reply:
thnx